library/nsxt_upgrade_postchecks.py

```diff
@@ -0,0 +1,169 @@
+                                  mgr_password, validate_certs, component_type, 
+                                  time_out=10800):
+  '''
+    params:
+    - endpoint: API endpoint.
+    - attribute_list: The attribute whose value should become the desired attribute value
+    - desired_attribute_value: The desired attribute value
+    
+    Function will wait till the attribute value derived from going deep to attribute list
+    becomes equal to desired_attribute_value.
+   '''
+  operation_time = 0
+  while True:
+    try:
+      (rc, resp) = request(manager_url + endpoint, headers=dict(Accept='application/json'),
+                           url_username=mgr_username, url_password=mgr_password, 
+                           validate_certs=validate_certs, ignore_errors=True)
+    except Exception as err:
+       pass
+    if resp.__contains__('results'):
+      flag = True
+      results = resp['results']
+      for result in results:
+        if result['post_upgrade_status']['status'] != 'COMPLETED'
+                             '' and result['type']= component_type.upper():
+          flag = False
+      if flag:
+        return None
+    time.sleep(15)
+    operation_time = operation_time + 15
+    if operation_time > time_out:
+      raise Exception('Operation timed out.')
+
+def main():
+  argument_spec = vmware_argument_spec()
+  argument_spec.update(timeout=dict(type='int', required=False),
+                      component_type=dict(required=True, choices=['mp', 'host', 'edge']))
+
+  module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+  state = module.params['state']
+  mgr_hostname = module.params['hostname']
+  mgr_username = module.params['username']
+  mgr_password = module.params['password']
+  validate_certs = module.params['validate_certs']
+  timeout = module.params['timeout']
+  component_type= module.params['component_type']
+
+  headers = dict(Accept="application/json")
+  headers['Content-Type'] = 'application/json'
+
+  mgr_hostname = get_upgrade_orchestrator_node(module, mgr_hostname, mgr_username, 
+                                            mgr_password, headers, validate_certs)
+
+  manager_url = 'https://{}/api/v1'.format(mgr_hostname)
+
+  #if state == 'present':
+  # Runs pre upgrade checks
+  if module.check_mode:
+    module.exit_json(changed=False, debug_out='Post upgrade checks will be executed.', 
+                     id='Pre upgrade checks')
+  try:
+    (rc, resp) = request(manager_url + '/upgrade/%s?action=execute_post_upgrade_'
+                        'checks' % component_type.upper(), data='', headers=headers,
+                        method='POST', url_username=mgr_username, 
+                        url_password=mgr_password, validate_certs=validate_certs, 
+                        ignore_errors=True)
+  except Exception as err:
+    module.fail_json(msg="Failed to execute post upgrade checks. Error[%s]." % to_native(err))
+
+  try:
+    if timeout is None:
```

Comment:
if time out is None, then set timeout =10800, don't need if else.

Reply:
Yeah, but we we always need to send timeout parameter. Can't send None value. Thus this check.